subprojects/dependency-management/src/main/java/org/gradle/internal/locking/EmptyLockConstraint.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright 2018 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.internal.locking;
+
+import org.gradle.api.artifacts.DependencyConstraint;
+import org.gradle.api.internal.artifacts.dsl.dependencies.LockConstraint;
+
+import java.util.Set;
+
+public class EmptyLockConstraint implements LockConstraint {
+
+    public static final EmptyLockConstraint INSTANCE = new EmptyLockConstraint();
+
+    public static LockConstraint getInstance() {
+        return INSTANCE;
+    }
+
+    private EmptyLockConstraint() {
+
+    }
+
+    @Override
+    public boolean isLockDefined() {
+        return false;
+    }
+
+    @Override
+    public Set<DependencyConstraint> getLockedDependencies() {
+        throw new UnsupportedOperationException("Cannot request locked dependencies when none are defined");
```

Comment:
:o: I feel like we can avoid this `EmptyLockConstraint` altogether. Could be a default lock constraint with an empty set of constraints, or something else, or maybe we don't need it. But it feels like accidental noise.

Reply:
Removed